keras_nlp/models/whisper/whisper_audio_feature_extractor.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.api_export import keras_nlp_export
+
+NUM_MELS = 80
+
+
+@keras_nlp_export("keras_nlp.models.WhisperAudioFeatureExtractor")
+class WhisperAudioFeatureExtractor(keras.layers.Layer):
+    """
+    Whisper audio feature extractor layer.
+
+    This layer takes in a batch of audio tensors, and computes the log-mel
+    spectrogram features for each audio tensor.
+
+    The input audio tensor can either be of shape `(length_of_audio,)` or
+    `(batch_size, length_of_audio)`. The output is a tensor of shape
+    `(batch_size, num_frames, NUM_MELS)`, where `num_frames` is
+    `(max_audio_length * sample_rate) / stride` and `NUM_MELS` is 80.
+
+    Args:
+        sample_rate: int, defaults to 16000. The sample rate of the audio.
+        num_fft_bins: int, defaults to 400. The size of the Fourier Transform in
+            STFT.
+        stride: int, defaults to 160. The distance between neighboring
+            sliding window frames while computing STFT.
+        max_audio_length: int, defaults to 30. The length of each audio chunk in
+            seconds. The input audio tensor will be padded/trimmed to
+            `max_audio_length*sample_rate`.
+
+    Examples:
+    ```python
+
```

Comment:
remove this top newline

Reply:
looks like this didn't happen?

Reply:
I think the comment was for an extra newline between ```python and the first line of code. We have a newline after Examples in all doc-strings.

Reply:
<img width="886" alt="image" src="https://github.com/keras-team/keras-nlp/assets/31360468/c91e5604-71ed-4bed-9193-cdac89c46b14">

Reply:
So, I have kept the newline after Examples, and removed the one after ```python.